### DISCUSSION TITLE
Reusable Vector Database

### DISCUSSION THREAD

**Discussion by chadbrouze** [ScrapeGraphAI/Scrapegraph-ai#268] [Q&A] [answered]
Hi everyone, Amazing and easy to use tool. Shoutout to Marco for making this!

Is there a way to reuse the vector database created on a graph run? I'd like to save tokens and time when querying the same website multiple times. Appreciate the help ,and I'd be happy to open a PR if this isn't already possible.

**Comment by itsperini** [comment, marked as answer]
Hey @chadbrouze right now the embedded documents is stored in the state and we are about to integrate caching functionalities so that you can reuse part of the graph :)

**Reply by chadbrouze** [reply]
Hey @PeriniM, has the caching functionality been integrated yet?

**Reply by itsperini** [reply]
@chadbrouze We are implementing [state persistence](https://burr.dagworks.io/concepts/state-persistence/) from Burr. This will allow to stop the scraping whenever you need and then restart from that point. Do you have some other suggestions?

**Reply by chadbrouze** [reply]
Interesting, I'll look into it and let you know (:
